Do not add Nest sensors via discovery

NextNebula opened the pull request:
**Description:**
Currently there is no way to disable the Nest sensors and binary sensors. This is caused by two problems.

1. Current behaviour is that all the sensors that are available for the Nest devices a user have (thermostat, protect or camera) are added as default when a Nest device is discovered by home-assistant and there is no config for the sensors. A user has no way to disable all sensors.
2. Discovered devices do not use a config so sensors that are added via discovery will not use the configured settings.

This PR changes the Nest sensors so that users have to configure it if they want to use them. Users need to add the sensors to there config. If configured without a monitored_condition list all available sensors will be added, otherwise a list can be specified.

**Example entry for `configuration.yaml` (if applicable):**
```yaml
# example with monitored_conditions list
sensor:
  - platform: nest
    monitored_conditions:
      - 'temperature'
      - 'target

# example without monitored_conditions list
binary_sensor:
  - platform: nest
```

**Checklist:**
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

pvizeli committed:
Migrate remote to async (#4678)

* Migrate remote to async

* add coro

* remove sync from init since only used in harmony

* import ATTR from remote

* remove unused sync stuff from tests

balloob committed:
Fix default auth influxdb (#4771)

balloob committed:
Fix Kodi auth (#4770)

technicalpickles committed:
Updated python-nest dependency (#4785)

* Updated python-nest dependency

This sha fixes two issues:

- min and max temperatures not being set when temperature isn't locked
- fixes error when setting farenheit with a .5

* gen requirements all

* Add fix for https://github.com/home-assistant/home-assistant/issues/4731

emlove committed:
Fix incorrect caching of /api/error_log (#4789)

balloob committed:
Update after service calls (#4795)

* Update after service calls

* Service update: wrap async_update in create_task

balloob committed:
Version bump to 0.34.4

balloob committed:
Merge pull request #4796 from home-assistant/release-0-34-4

0.34.4

Commit:
Fix python-nest version bump (#4799)

* Fix python-nest version bump

* Change SHA so version in HASS and dependency are the same

technicalpickles committed:
Bump python-nest to fix issue with Nest Cam without activity zones (#4820)

* Bump python-nest to fix issue with Nest Cam without activity zones

* bump to include fix python-nest dependency with hvac_state

* regenerate requirements_all.txt

NextNebula committed:
Fix do not add Nest sensors via discovery

NextNebula committed:
Fix do not add Nest sensors via discovery

NextNebula committed:
Fix assign SENSOR_TYPES_DEPRECATED twice

NextNebula committed:
Fix assign SENSOR_TYPES_DEPRECATED twice

NextNebula committed:
Fix lint errors

technicalpickles commented on homeassistant/components/binary_sensor/nest.py:
```diff
@@ -60,11 +61,8 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup Nest binary sensors."""
-    if discovery_info is None:
```
This is actually needed. Without this, the configurator never shows up https://github.com/home-assistant/home-assistant/issues/4730

R1chardTM replied:
The configurators shows up and works as intended. But I see now my code introduces two problems. Both sensors load with a error at startup if Nest is not configured. And a restart is needed to load the sensors after configuration is complete.

technicalpickles commented on homeassistant/components/binary_sensor/nest.py:
```diff
@@ -50,7 +50,8 @@
         vol.All(vol.Coerce(int), vol.Range(min=1)),
-    vol.Required(CONF_MONITORED_CONDITIONS):
+    vol.Optional(CONF_MONITORED_CONDITIONS,
+                 default=_VALID_BINARY_SENSOR_TYPES):
```
Nice, I had hoped there was a way to do this but never around to finding it 😁

technicalpickles commented on homeassistant/components/nest.py:
```diff
@@ -88,8 +88,6 @@ def setup_nest(hass, nest, config, pin=None):
 
     _LOGGER.debug("proceeding with discovery")
     discovery.load_platform(hass, 'climate', DOMAIN, {}, config)
-    discovery.load_platform(hass, 'sensor', DOMAIN, {}, config)
```
Nest Protects are _only_ sensors, so Nest Protect users would have to explicitly configure the sensor, which is what I was trying to avoid.

technicalpickles reviewed (commented):
> Current behaviour is that all the sensors that are available for a Nest device (thermostat, protect or camera) 

That's not exactly correct. Only sensors that are appropriate to the device are added. There's logic to check the device type, and add only fields it supports. If that isn't happening, it's definitely a bug

R1chardTM commented:
> That's not exactly correct. Only sensors that are appropriate to the device are added. There's logic to check the device type, and add only fields it supports. If that isn't happening, it's definitely a bug.

That is what I mean. You indeed only get the sensors for the devices you have. But it is not possible to configure that only specific or no sensors show up for the devices you have. As discussed on Gitter I will change this PR so the code is both user friendly for users that do not want sensors and users that have a Nest protect that only works with sensors.

kellerza commented on homeassistant/components/binary_sensor/nest.py:
```diff
@@ -60,11 +61,8 @@
-        return
-
     nest = hass.data[DATA_NEST]
-    conf = config.get(CONF_MONITORED_CONDITIONS, _VALID_BINARY_SENSOR_TYPES)
```
I'm pretty sure in the past this default would always have been used -- Using load_platform, config is *never* passed to the setup_platform method - see [here](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_component.py#L83)

So the PR's statement 2 is because of the config Not being passed in...

The while idea if having validated schema under the platform AND using discovery only is conflicting in the current nest implementation

The intended design for load_platform was always to have the user config only on the main component, not the platforms! So I would suggest moving the config to the component level rather and using load_platform appropriately

The return if None statement should as a matter of fact fraise a warning stating to not do the config ont he platform level

kellerza reviewed (changes requested):
Existing implementation is not correct, but suggest that the PR changes it into using load_platform correctly, not removing load_platform

Danielhiversen committed:
Configurable scan options for nmap (#4838)

stefan-jonasson committed:
Fixed crash during light objects initizilation (#4835)

* Fixed crash when lights objects was inited

lwis committed:
[device_tracker.gpslogger] Add additional activity attribute.

Danielhiversen committed:
Merge pull request #4846 from lwis/gpslogger

[device_tracker.gpslogger] Add additional activity attribute.

lwis committed:
[device_tracker] Don't clear GPS coordinates if no GPS seen (#4848)

kellerza committed:
Revert "[device_tracker] Don't clear GPS coordinates when using two device trackers." (#4851)

pvizeli committed:
Migrate alarm control panel to async (#4807)

* Merge alarm control panel to async

* fix lint

dcrypt3d committed:
Ecobee detect Smart Away (#4769)

* Ecobee autoAway Event

* Update ecobee.py

Checking if event['running'] true is pointless because if false event['type'] will equal template and when true type will only be 'hold' or 'autoAway' so I've removed this check from the statement

Jypy committed:
Netatmo improving Battery info (#4724)

* Improving Battery info

Improving battery status info (to reflect NetAtmo API documentation and change deprecated DeviceList for WeatherStationData

* Fixes from previous update

Fix the hound issue.
add battery_lvl, WindAngle_value, GustAngle_value, rf_status_lvl, wifi_status_lvl

Danielhiversen committed:
vlc media player (#4800)

* vlc media player

* Update vlc.py

tchellomello committed:
Added support  to language codes on Weather Underground (#4815)

* Added supported to language codes to Weather Underground

* Removed unecessary None assigments

Commit:
adding sensehat plugin (#4775)

* adding sensehat plugin

* added

* fix PR

* requirement updated

* Update sensehat.py

jawilson committed:
Use current mode to determine which temperature attributes to use (#4858)

balloob committed:
Fix Plex from doing I/O inside event loop (#4857)

balloob committed:
Fix Plex from doing I/O inside event loop (#4857)

balloob committed:
Fix Nest doing I/O inside event loop (#4855)

balloob committed:
Fix Nest doing I/O inside event loop (#4855)

DavidLP committed:
Fix config validation (#4853) (#4854)

michaelarnauts committed:
Add support for Hue LightGroups (#4744)

* Add support for Hue LightGroup entity

* Don't filter on LightGroup and add properties for a group

* Reuse code from HueLight in HueLightGroup

* Remove HueLightGroup and add is_group variable to HueLight

* Make linter happy

* Update light or lightgroup state when a new state is available

* Use schedule_update_ha_state() to schedule the state update. Drop new_lightgroups and use new_lights instead.

* code style fix

glance- committed:
Improve denon media_player (#4836)

* Add debug level logging of messages in denon

* Added media stop for Denon AVR Media Player

* Sort source list

* Rework input selection for Denon AVR

This reworks the input selection, adding more modes and making it so
that the media controls are only announced in modes where they actually
makes sense.

* Added real media info for Denon AVR Media modes

* Read more configuration from denon devices

This reads network name, and overrides the local name with that.

This also reads the source names and reconfigures the input list to
those names, and also reads the source deleted list and removes the
inputs that are set to deleted in the device.

* Discover and handle max volume in Denon media player

* Rework source discovery in Denon media player

This uses SSFUN as authorative source for which sources that we should
present.

molobrakos committed:
Telldus Live: (#4645)

- Implemented support for covers and dimmable lights.
- Removed global object, use hass.data.
- Disabled polling via update.
- Inherit from common TelldusLiveEntity device.
- Configurable polling interval
- Use https API endpoint
- Use tellduslive package

mezz64 committed:
Add media position support and trailer type to Emby (#4792)

* Add media position support and trailer type to Emby

* Adjustments to mitigate TypeError

* Simplify media_position property

* Update handling when data isn't available

* Update emby.py

emlove committed:
Flic: Support use of queued events within timeout (#4822)

* Flic: Support use of queued events within timeout

* Linter fixes

tchellomello committed:
Added resolution support to Amcrest cameras (#4860)

* Added resolution support to Amcrest cameras

* Ordered alphabetically DEFAULT_ options

balloob committed:
Version bump to 0.34.5

balloob committed:
Merge pull request #4862 from home-assistant/release-0-34-5

0.34.5

balloob committed:
Merge branch 'dev' into master

balloob committed:
Merge pull request #4863 from home-assistant/master

Backmerge point releases

molobrakos committed:
[tellduslive] Upgrade requirement (#4865)

molobrakos committed:
Display error message instead of exception (#4866)

* Display error message instead of exception

Display error message in log instead of stack trace.
(Usually happens when a server is already running at the same port.)

* Update __init__.py

Better error handling when reading SSL certificate

* Update __init__.py

* Update __init__.py

balloob committed:
Fix dev tag detection in release script (#4873)

mezz64 committed:
Add Hikvision binary sensor component (#4825)

* Add Hikvision binary sensor component

* Simplify customize configuration

* Add delay attribute

* Remove use of threading timer, fix delay functionality

pvizeli committed:
TTS Component / Google speech platform (#4837)

* TTS Component / Google speech platform

* Change file backend handling / cache

* Use mimetype / rename Provider function / allow cache on service call

* Add a memcache for faster response

* Add demo platform

* First version of unittest

* Address comments

* improve error handling / address comments

* Add google unittest & check http response code

* Change url param handling

* add test for other language

* Change hash to sha256 for same hash on every os/hardware

* add unittest for receive demo data

* add test for error cases

* Test case load from file to mem over aiohttp server

* Use cache SpeechManager level, address other comments

* Add service for clear cache

* Update service.yaml

* add support for spliting google message

auduny committed:
Remove libtelldus-core-dev from Dockerfile (#4878)

Remove unnecessary  libtelldus-core-dev from Dockerfile . Ref https://github.com/home-assistant/home-assistant/pull/4680#issuecomment-266006310

pvizeli committed:
Migrate REST switch to async (#4517)

* Migrate REST switch to async

* Update rest.py

* Address comments from paulus

molobrakos committed:
Update __init__.py (#4877)

Cleaner exit by not throwing exception if server was not set during initialization of component (ref https://github.com/home-assistant/home-assistant/pull/4866)

NextNebula committed:
Move Nest sensor config to Nest component

NextNebula committed:
Merge branch 'nest-sensors' of https://github.com/R1chardTM/home-assistant into nest-sensors

R1chardTM commented:
Somehow I destroyed my branch with a merge. Will make a new PR when i got this worked out.

kellerza commented:
You should still be able to save your branch, see [here](https://home-assistant.io/developers/development_catching_up/)

```
git fetch --all
git rebase updstream/dev
```
